packages/@aws-cdk/aws-cognito-identitypool/lib/identitypool.ts

```diff
@@ -0,0 +1,504 @@
+      allowClassicFlow: props.allowClassicFlow,
+      identityPoolName: this.physicalName,
+      developerProviderName: authProviders.customProvider,
+      openIdConnectProviderArns,
+      samlProviderArns,
+      supportedLoginProviders,
+      cognitoIdentityProviders: Lazy.any({ produce: () => this.cognitoIdentityProviders }),
+    });
+    this.identityPoolName = cfnIdentityPool.attrName;
+    this.identityPoolId = cfnIdentityPool.ref;
+    this.identityPoolArn = Stack.of(scope).formatArn({
+      service: 'cognito-identity',
+      resource: 'identitypool',
+      resourceName: this.identityPoolId,
+      arnFormat: ArnFormat.SLASH_RESOURCE_NAME,
+    });
+    this.authenticatedRole = props.authenticatedRole ? props.authenticatedRole : this.configureDefaultRole('Authenticated');
+    this.unauthenticatedRole = props.unauthenticatedRole ? props.unauthenticatedRole : this.configureDefaultRole('Unauthenticated');
+    const attachment = new IdentityPoolRoleAttachment(this, 'DefaultRoleAttachment', {
+      identityPool: this,
+      authenticatedRole: this.authenticatedRole,
+      unauthenticatedRole: this.unauthenticatedRole,
+      roleMappings: props.roleMappings,
+    });
+    attachment.node.addDependency(this);
+  }
+
+  /**
+   * Add a User Pool to the IdentityPool and configure User Pool Client to handle identities
+   */
+  public addUserPoolAuthentication(userPool: IUserPoolAuthenticationProvider): void {
+    const providers = this.configureAuthenticationProviders(userPool);
+    providers.forEach(provider => this.cognitoIdentityProviders.push(provider), this);
+  }
+
+  /**
+   * Adds Role Mappings to Identity Pool
+  */
+  public addRoleMappings(...roleMappings: IdentityPoolRoleMapping[]): void {
+    if (!roleMappings || !roleMappings.length) return;
+    this.roleAttachmentCount++;
+    const name = 'RoleMappingAttachment' + this.roleAttachmentCount.toString();
+    const attachment = new IdentityPoolRoleAttachment(this, name, {
+      identityPool: this,
+      authenticatedRole: this.authenticatedRole,
+      unauthenticatedRole: this.unauthenticatedRole,
+      roleMappings,
+    });
+    attachment.node.addDependency(this);
+  }
+
+  /**
+   * Configure Authentication Providers for User Pools
+   */
+  private configureAuthenticationProviders(provider: IUserPoolAuthenticationProvider): CfnIdentityPool.CognitoIdentityProviderProperty[] {
+    const identityProviders = provider.identityProviders || [];
+    return identityProviders.map(identityProvider => {
+      return {
+        clientId: provider.clientId,
+        providerName: identityProvider.providerName,
+        serverSideTokenCheck: !provider.disableServerSideTokenCheck,
+      };
+    });
+  }
+
+  /**
+   * Configure Default Roles For Identity Pool
+   */
+  private configureDefaultRole(type: string): IRole {
+    const assumedBy = this.configureDefaultGrantPrincipal(type.toLowerCase());
+    const role = new Role(this, `${this.id}${type}Role`, {
```

Comment:
```suggestion
    const role = new Role(this, `${type}Role`, {
```

Don't include the `id`